.github/workflows/python-package.yml

```diff
@@ -25,19 +29,21 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip
-          pip install black flake8 pytest coverage
```

Comment:
Because in `setup.py`, they're already defined as the dependency. So installing with `pip install '.[test]'` will already install them. It's a good practice to put dependencies in one place rather than having them in multiple places.